src/openssl_stream.c

```diff
@@ -243,7 +243,11 @@ int openssl_connect(git_stream *stream)
 		return ssl_set_error(st->ssl, ret);
 	}
 
-	if ((ret = SSL_connect(st->ssl)) <= 0)
+	do {
+		ret = SSL_connect(st->ssl);
+	} while (SSL_get_error(st->ssl, ret) == SSL_ERROR_WANT_READ || SSL_get_error(st->ssl, ret) == SSL_ERROR_WANT_WRITE);
```

Comment:
I can't say as the SSL docs don't provide enough info. It's however the same pattern as other places in the code where an SSL library function returns `SSL_ERROR_WANT_READ` or `SSL_ERROR_WANT_WRITE`. So if it can "burn the CPU" here, it can elsewhere too, and any solution, if there's a problem, would need to be global.